src/cosim/observer/file_observer.cpp

```diff
@@ -448,12 +448,16 @@ file_observer::simulator_logging_config file_observer::parse_config(const std::s
                             const auto variableDescription =
                                 find_variable(simulator->model_description(), name);
 
-                            switch (variableDescription.type) {
+                            if (!variableDescription) {
+                                throw std::invalid_argument("Can't find variable descriptor with name " + name + " for model with name " + simulator->model_description().name);
```

Comment:
This shouldn't be an `invalid_argument` exception, as that indicates a programming error. (It was the wrong type to use before too, but even more so now.)

Ideally, we should add a (somewhat re-usable) error code like `name_not_found` to `cosim::errc` and then throw a `cosim::error` with this code. However, this would require changes in both `exception.hpp` and `exception.cpp`, and we're not consistent in our use of this exception type as it is. So I'm satisfied if `invalid_argument` just gets changed to `runtime_error`.